docs/src/pages/ComponentArchitecture.mdx

```diff
@@ -0,0 +1,97 @@
+- Consumers should not be penalized for code they don't use (through size, perf, etc).
+- Don't overgeneralize until we have a clear need (abstracting should solve a bundle size goal, or DRY goal, or partner use case)
+
+# Base components
+
+Each component should be built upon a "base component".
+
+In order to keep sorted lists formatted neatly, each base component should be named with the form: `ComponentNameBase.tsx`
+
+#### Applications which load their own CSS in other means should not pay the penalty of loading a runtime CSS-in-JS package, especially if the class names are known at time of authoring.
+
+Base components should expect a prop called `classes`, which is either a
+dictionary of keys to classnames (where a key represents a portion of the DOM), or
+a callback which will return the same dictionary when called with the component's
+state.
+
+#### Applications which want to override specific aspects of a component should not incur a penalty of loading a concretely referenced component that will not be rendered in the final output.
+
+Base components should expect a prop called `slots`, which is a dictionary of element names mapped
+to their respective concrete implementation. The base component should default to rendering native DOM
+elements for each slot unless an override is passed.
+
+#### Any external dependencies mean negative parsing time, bundle size and runtime impact for consumers that don't intend to use them.
+
+Base components should get all outside information exclusively from `props`.
+
+#### Hooks provide a desirable level of abstraction between business logic and markup. Reusable hooks further provide consumers with the ability to reuse smaller pieces of library controls.
+
+Hooks should be created to manage internal state of base components. For sets
+of functionality common to several components, shared hooks should be abstracted
+and exported from a common platform package.
+
+#### There is a limited set of accessibility options that work well together, and each combination should be tested cohesively. Further, accessibility is closely tied to a components internal state, and they are likely to iterate concurrently.
+
+Each base component with accessibility concerns should provide its own implementation of accessibility props and handlers.
+For variants that are expected in the overall component library, state hooks should be reused and a new component simply
+created with a different use-case.
+
+It is important to have consistent accessibility, and passing in a new accessibility model might create major
+challenges to keep our components accessible, and also predictable. If a particular component needs to have multiple
+accessibility models (keyboarding, attributes), it should make that decision based on a single prop with a fixed set
+of implementation options (i.e. a `accessibilityMode` prop that accepts a fixed set of string values as input).
+
+# Themed components
```

Comment:
A few more points that should be mention regarding theming, based on the client's requirements we have had so far:

# Theming

The theming mechanism as part of Fluent should be flexible and provide support for as many scenarios as possible (CSS stylesheets, as well as CSSinJS solution that will not tight the users to just one CSS in JS framework).

#### A component should be able to render any theme (including tokens, styles, icons and animations)

The client should be able to reuse the components between different themes. Supporting new theme in the application should not requre building new components.

If new variants are added in the new theme, then the clients should be able to compose the existing component, by adding new design terms (via the props API).

#### Themes should cascade and merge allowing subtree themes to be updated at runtime and have the ability to adjust the incoming theme.

If some parameters/tokens are changed on the root of the application, the changes should be propagated to all nested subtrees.

#### RTL mode should be configurable for the entire application or a subtree of the application.

It is important to allow the users to specify the RTL/LTR mode for the entire application once, but if necessary they can specify it only in a subtree of the application (for example one ChatMessage)